### PR TITLE
change robot drive forward algo

### PR DESCRIPTION
**Pull request by elinliu2**

### CODE REVIEW

File: jetson/src/core/robot.cpp [hunk 1]
```diff
@@ -73,34 +73,32 @@ void Robot::updateArmPosition() {
...
 
 	while (delta.mag() < ROBOT_TOL) {
-
-		double r = calculate_radius(delta.x, delta.y);
-
 		// TODO: how to figure ideal v?
 		double v = IDEAL_LINEAR_SPEED;
-		double w = v / r;
+
+		double goal_orientation = atan2(delta.y, delta.x);
```
Comment (reubenkw) [inline jetson/src/core/robot.cpp]: isn't this backwards? `atan 0 = 0` should be the case of `delta.x = 0` and `delta.y` be the distance the robot moves forward

File: jetson/src/core/robot.cpp [hunk 2]
```diff
@@ -73,34 +73,32 @@ void Robot::updateArmPosition() {
...
+            
+		double error_angular = std::abs(std::fmod(goal_orientation - robotAngle, M_PI));
+
+		double w = angular_error_gain * error_angular;
```
Comment (reubenkw) [inline jetson/src/core/robot.cpp]: mod PI and abs?? Im I dumb. How is this gonna work. I think you want the error to be `std::fmod(goal_orientation - robotAngle, 2*M_PI) - M_PI` such that it is between -PI and PI.

File: jetson/src/core/robot.cpp [hunk 3]
```diff
@@ -73,34 +73,32 @@ void Robot::updateArmPosition() {
...
+void Robot::driveRobotForward(Point2D goal) {
+	double angular_error_gain = 2;
 	robotPosition = Point2D(0, 0);
 	drive[frontLeft].resetElapsedDistance();
 	drive[backLeft].resetElapsedDistance();
 	drive[frontRight].resetElapsedDistance();
 	drive[backRight].resetElapsedDistance();
+	Point2D delta = goal - robotPosition;
 
 	while (delta.mag() < ROBOT_TOL) {
```
Comment (reubenkw) [inline jetson/src/core/robot.cpp]: Not sure what suggestion to make but I'd like to just acknowledge that an overshoot would result in a runaway robot. Or at least a spiraling robot

**Review by reubenkw — CHANGES_REQUESTED** [review]

File: jetson/src/core/robot.cpp (hunk 1 above)
Comment (elinliu2) [reply jetson/src/core/robot.cpp]: agreed but then i think it should be -x for RHR convention

**Review by reubenkw — APPROVED** [review]

**Pull request merged**